db.go

```diff
@@ -690,6 +690,61 @@ type Result interface {
 	// using All().
 	All(sliceOfStructs interface{}) error
 
+	// Paginate splits the results of the query into pages containing pageSize
+	// items.  When using pagination previous settings for Limit and Offset are
+	// ignored.
+	//
+	// Use Page() to define the specific page to get results from.
+	//
+	// Example:
+	//
+	// r = q.Paginate(12)
+	Paginate(pageSize uint) Result
+
+	// Page makes the result set return results only from the page identified by
+	// pageNumber. Page numbering starts from 0.
+	//
+	// Example:
+	//
+	// r = q.Paginate(12).Page(4)
+	Page(pageNumber uint) Result
+
+	// Cursor defines the column that is going to be taken as basis for
+	// cursor-based pagination.
+	//
+	// Example:
+	//
+	// a = q.Paginate(10).Cursor("id")
+	// b = q.Paginate(12).Cursor("-id")
+	//
+	// You can set "" as cursorColumn to disable cursors.
+	Cursor(cursorColumn string) Result
+
+	// NextPage returns the next results page according to the cursor. It expects
+	// a cursorValue, which is the value the cursor column had on the last item
+	// of the current result set.
+	//
+	// Example:
+	//
+	// current = current.NextPage(items[len(items)-1].ID)
+	NextPage(cursorValue interface{}) Result
```

Comment:
We need a parameter to define where to start counting elements for the next page (lower bound), we can't store any state within the result. Same with PrevPage, in that case we need the user to provide an upper bound.